### DISCUSSION TITLE
Is the "Get List" function of hfs2 still in hfs3?

### DISCUSSION THREAD

**Discussion by Dnxi** [rejetto/hfs#231] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/61756759/235196118-48f8a250-5860-4da2-ab4a-3343260cc43a.png)

**Comment by rejetto** [comment]
Nope. Do you need it?

**Reply by Dnxi** [reply]
Yes, it would be very useful for me

**Comment by rejetto** [comment]
Ok. I'm thinking about where to have it.
 Another button on top doesn't fit mobiles. 
 Options dialog could be an easy one, but I'm not totally sure it makes sense. 
Clicking the folder name in the breadcrumbs now is for reload, but could bring you the current folder menu. Cool, but not the easiest to guess.

**Comment by rejetto** [comment]
By the way, can you tell me what kind of use do you make of it?

**Comment by Dnxi** [comment]
![image](https://user-images.githubusercontent.com/61756759/235213516-a67faa6a-3e1c-4a66-a103-67c08f7db2a6.png)
I want to use it for an app, I really just need to be able to access via url

**Reply by rejetto** [reply]
Ok. I asked because the more I know about what the users do, the more I can tune my work to better fit, so any detail you may add is good, but not necessary

**Comment by rejetto** [comment, marked as answer]
please, give a look and tell me if it fits your needs: https://github.com/rejetto/hfs/discussions/241

<img width="234" alt="image" src="https://user-images.githubusercontent.com/1367199/236704786-2663fc87-9f09-49dd-b93c-00c7cf28c4eb.png">

**Reply by Dnxi** [reply]
How do I start HFS 0.45.0?
I use `npx -y hfs@latest`

**Reply by rejetto** [reply]
Ok, that's cool, I was wondering if anyone was using this way. 😄 I will work to make the beta available for you as well, and let you know

**Comment by rejetto** [comment]
ok, now i published it for npx, but it's in the "beta" channel, to avoid unwanted updates.
To try new version, my best advice for the moment is to change `npx -y hfs@latest` to `npx -y hfs@beta`

I would like to find a way to specify both latest and beta, as otherwise you'll have to change it back when official is published.

**Comment by rejetto** [comment]
I noticed that on Windows (is it you?), npx was saving to the wrong folder.
Latest version is fixing this, but before running it you may need to move your files from C:\users\NAME
to C:\users\NAME\\.hfs

otherwise you may see the program start "blank"
sorry

**Reply by Dnxi** [reply]
Don't worry, I use Ubuntu (Linux)
But there would be some parameter in the URL 
![http://files.example.com:8080/files/?get=list&folders=*&depth=*&](https://user-images.githubusercontent.com/61756759/236865584-ca2ef94e-0a6a-4134-9126-6ba05e01773d.png)
to display the full URL of the files in the list 
![http://files.example.com:8080/folder1/folder2/file.txt](https://user-images.githubusercontent.com/61756759/236864174-6cfb5c41-c85c-416d-bfdf-00bad8ad0285.png)

**Reply by rejetto** [reply]
ok

**Reply by rejetto** [reply]
I already have the solution for your request, but while working I noticed that "deep" listings (even search) can stop randomly.
I'm working on this problem now

**Reply by rejetto** [reply]
ok, you can update your beta now.

You'll see that the solution I used for the missing prefix is to specify it in the URL.
While dumb and longer, it is always the right URL. There are in fact edge cases where making the right URL can be tricky, like reverse-proxies.
I have to think if I can use another thing.

**Reply by rejetto** [reply]
well, since not having a rev-proxy is quite common, I suppose I can introduce a default mode where the 'prepend' is not specified and just guessed

**Reply by Dnxi** [reply]
You can also use what the user specifies
![image](https://user-images.githubusercontent.com/61756759/237063687-3dc1ebdb-45fe-4a4d-81d3-c50dae415171.png)

**Reply by rejetto** [reply]
yes, that's exactly what i'm going to do, but also be able to manage the case where user didn't configure that.
I already have it, just didn't publish yet

**Reply by skbohrer** [reply]
This is an old thread, but I have just started looking at v3, after being a v2 user for an embedded app long ago.

With HFS version 2, my app could fetch the plain text "list" version by appending "~files.lst" to the URL. With v3, seems like the append string is  "?get=list". Any simple way to change it for backwards compatibility, or should I redo the app? (Also, v2 allowed the list output format to be specified with a template, to give filename, size, and mod-date as CSV text.) Probably the cleanest answer is to re-code the app to read the html output directly. Years ago, the list format saved me from having to do that.

**Reply by rejetto** [reply]
this deserved a separate thread.
if you want to have other info about files, your best options is to use the same API (almost) the frontend of HFS uses.
curl -v "http://localhost/~/api/get_file_list?uri=%2Ffolder"

try something like this, and you'll get a JSON object, with a "list" property.
of course you have to adapt the "uri" parameter with an url-encoded version of the path you are interested in.

**Reply by skbohrer** [reply]
Thanks for the reply! New version seems very slick so far, especially the auto-magic SSL cert setup for https! I am looking through my app to see the commands it used with the old HFS server, and then I can update them to use the new versions.

One question is for the format of the list output : On the old version we could specify that, through a template file I believe (but I can't actually find my example of my old template!)

**Reply by rejetto** [reply]
each entry of the list is an object with these fields
n: string, name of the entry, including relative path when searched in sub-folders.
s: number, size of the entry, in bytes. It may be missing, for example for folders.
m: string, Date, modified-time, optional

**Reply by rejetto** [reply]
i forgot i had it documented
https://hfs-3.apidog.io/api-6834191
